tests/test_size.py

```diff
@@ -56,56 +59,190 @@ def bar(self):
+def test_size_file():
     example = """
-while "b" > "a":
-    pass
-else:
-    pass
+class Foo:
+    with open("/dev/null") as f:
+        f.write("")
 """
     check_size(example, 3)
 
 
+def test_size_try_except_else():
+    example = """
+class Foo:
+    try:
+        x = sys.argv[99]
+    except IndexError:
+        pass
+    except Exception:
+        pass
+    else:
+        pass
+"""
+    check_size(example, 9)
+
+
 def test_size_try_finally():
     example = """
-try:
-    1/0
-finally:
-    return 99
+class Foo:
+    try:
+        1/0
+    finally:
+        return 99
+"""
+    check_size(example, 5)
+
+
+def test_size_try_except():
+    example = """
+class Foo:
+    try:
+        foo()
+    except:
+        bar()
+"""
+    check_size(example, 5)
+
+
+def test_size_try_excepts():
+    example = """
+class Foo:
+    try:
+        foo()
+    except IOError:
+        bar()
+    except AttributeError:
+        pass
+"""
+    check_size(example, 7)
+
+
+def test_size_for():
+    example = """
+class Foo:
+    for i in range(10):
+        print(i)
 """
     check_size(example, 3)
 
 
 def test_size_for_else():
     example = """
-for arg in sys.argv:
-    print("loop")
-else:
-    print("else")
+class Foo:
+    for arg in sys.argv:
+        print("loop")
+    else:
+        print("else")
+"""
+    check_size(example, 5)
+
+
+def test_size_class_nested():
+    example = """
+class Foo:
+    class Bar:
+        pass
+"""
+    check_size(example, 3)
+
+
+def test_size_multi_line_return():
+    example = """
+class Foo:
+    def long_string_return(o):
+        return (
+            'very'
+            'long'
+            'string')
+"""
+    # We currently cannot handle code ending with multiline statements
+    check_size(example, 3)
+
+
+def test_size_comment_after_last_line():
+    example = """
+class Foo:
+    def bar():
+        # A comment.
+        pass
+        # This comment won't be detected.
+"""
+    check_size(example, 4)
+
+
+@pytest.mark.skipif(not PY35, reason='requires Python3.5')
+def test_size_generator():
+    example = """
+class Foo:
+    def bar():
+        yield something
+"""
+    check_size(example, 3)
+
+
+@pytest.mark.skipif(not PY35, reason='requires Python3.5')
+def test_size_coroutines_basic():
+    example = """
+class Foo:
+    async def foo(some_attr):
+        pass
+"""
+    check_size(example, 3)
+
+
+@pytest.mark.skipif(not PY35, reason='requires Python3.5')
+def test_size_await():
+    example = """
+class Foo:
+    async def __aenter__(self):
+        await log('entering context')
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await log('exiting context')
+"""
+    check_size(example, 6)
+
+
+@pytest.mark.skipif(not PY35, reason='requires Python3.5')
+def test_size_async_with():
+    example = """
+class Foo:
+    async def bar():
+        async with x:
+            pass
+"""
+    check_size(example, 4)
+
+
+@pytest.mark.skipif(not PY35, reason='requires Python3.5')
```

Comment:
Can you define a decorator `skip_if_not_has_async` once to avoid repeating the code?